packages/mongodb-rag-core/src/contentStore/MongoDbEmbeddedContentStore.ts

```diff
@@ -95,6 +95,68 @@ export function makeMongoDbEmbeddedContentStore({
       return await embeddedContentCollection.find(pageIdentity(page)).toArray();
     },
 
+    async getPagesFromEmbeddedContent({
+      dataSources,
+      chunkAlgoHash,
+      inverseChunkAlgoHash = false,
+    }): Promise<PersistedPage[]> {
+      const pipeline = [
+        {
+          $match: {
+            ...(dataSources ? { sourceName: { $in: dataSources } } : undefined),
+            chunkAlgoHash: inverseChunkAlgoHash
+              ? { $ne: chunkAlgoHash }
+              : chunkAlgoHash,
+          },
+        },
+        {
+          $lookup: {
+            from: "pages",
```

Comment:
one potential issue w/ this is that the `PageStore` collection might not be named `Pages`. also by not passing the `PageStore`, you're coupling this method to MongoDB and the 2 collections existing in the same, which shouldn't necessarily be assumed.

for this reason, an implementation more like the one discussed in my comment here could be preferable https://github.com/mongodb/chatbot/pull/580#discussion_r1887467877